src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_gemm_emitter.cpp

```diff
@@ -62,12 +80,34 @@ void jit_gemm_emitter::emit_impl(const std::vector<size_t>& in, const std::vecto
     Xbyak_aarch64::XReg x1(1);
     Xbyak_aarch64::XReg x2(2);
     Xbyak_aarch64::XReg x3(3);
-    h->str(Xbyak_aarch64::XReg(in[0]), pre_ptr(h->sp, -get_vec_length()));
-    h->str(Xbyak_aarch64::XReg(in[1]), pre_ptr(h->sp, -get_vec_length()));
-    h->str(Xbyak_aarch64::XReg(out[0]), pre_ptr(h->sp, -get_vec_length()));
-    h->ldr(x3, post_ptr(h->sp, get_vec_length()));
-    h->ldr(x2, post_ptr(h->sp, get_vec_length()));
-    h->ldr(x1, post_ptr(h->sp, get_vec_length()));
+    Xbyak_aarch64::XReg aux_reg(5);
+
+    // Prepare memory pointers with offsets
+    std::vector<size_t> mem_ptrs_idxs{in[0], in[1], out[0]};
+    const auto& mem_ptrs = utils::transform_idxs_to_regs(mem_ptrs_idxs);
+
+    // Apply memory offsets and load adjusted pointers
+    std::vector<Xbyak_aarch64::XReg> load_regs{x1, x2, x3};
+
+    // Dynamically choose safe auxiliary registers that don't conflict with mem_ptrs or load_regs
+    std::vector<size_t> used_indices;
+    for (const auto& reg : mem_ptrs)
+        used_indices.push_back(reg.getIdx());
+    for (const auto& reg : load_regs)
+        used_indices.push_back(reg.getIdx());
+    used_indices.push_back(0);  // X0
+    used_indices.push_back(9);  // X9 - func_reg
+
+    std::vector<Xbyak_aarch64::XReg> aux_regs;
+    for (size_t i = 2; i <= 30 && aux_regs.size() < 3; ++i) {
+        if (std::find(used_indices.begin(), used_indices.end(), i) == used_indices.end()) {
+            aux_regs.emplace_back(static_cast<int>(i));
+        }
+    }
```

Comment:
You have already implemented the helper `get_aux_gpr`, so we can add one more helper `get_aux_gprs` which return std vector 😄
And then call the new helper in the both emitters

```suggestion
    std::vector<size_t> used_indices;
    for (const auto& reg : mem_ptrs)
        used_indices.push_back(reg.getIdx());
    for (const auto& reg : load_regs)
        used_indices.push_back(reg.getIdx());

    std::vector<Xbyak_aarch64::XReg> aux_regs = get_aux_gprs(used_indices);
```

Reply:
Ah, yes, you're right. Applied for both